test/CompoundPrizePool.test.js

```diff
@@ -0,0 +1,260 @@
+  
+        // expect finish
+        await comptroller.mock.afterWithdrawWithTimelockFrom.withArgs(wallet._address, toWei('10'), ticket.address).returns()
+  
+        // setup timelocked withdrawal
+        await prizePool.withdrawWithTimelockFrom(wallet._address, toWei('10'), ticket.address)
+  
+        expect(await prizePool.timelockBalanceOf(wallet._address)).to.equal(toWei('10'))
+        expect(await prizePool.timelockBalanceAvailableAt(wallet._address)).to.equal('10')
+        expect(await prizePool.timelockTotalSupply()).to.equal(toWei('10'))
+      })
+    })
+  
+    describe('sweepTimelockBalances()', () => {
+      it('should do nothing when no balances are available', async () => {
+        // updateAwardBalance
+        await cToken.mock.balanceOfUnderlying.returns('0')
+        await ticket.mock.totalSupply.returns('0')
+  
+        // now execute timelock withdrawal
+        await expect(prizePool.sweepTimelockBalances([wallet._address]))
+          .not.to.emit(prizePool, 'TimelockedWithdrawalSwept')
+          .withArgs(wallet._address, wallet._address, toWei('10'))
+      })
+  
+      it('should sweep only balances that are unlocked', async () => {
+        // updateAwardBalance
+        await cToken.mock.balanceOfUnderlying.returns('0')
+        await ticket.mock.totalSupply.returns('0')
+  
+        // force current time
+        await prizePool.setCurrentTime('1')
+  
+        // expect ticket burns from both
+        await ticket.mock.controllerBurnFrom.returns()
+  
+        debug('GOT HERE 1')
+  
+        // withdraw for a user, and it's eligible at 10 seconds
+        await comptroller.mock.calculateWithdrawalUnlockTimestamp.returns(10)
+        await comptroller.mock.afterWithdrawWithTimelockFrom.withArgs(wallet._address, toWei('11'), ticket.address).returns()
+        await prizePool.withdrawWithTimelockFrom(wallet._address, toWei('11'), ticket.address)
+  
+        debug('GOT HERE 2')
+  
+        // withdraw for a user, and it's eligible at 20 seconds
+        await comptroller.mock.calculateWithdrawalUnlockTimestamp.returns(20)
+        await comptroller.mock.afterWithdrawWithTimelockFrom.withArgs(wallet2._address, toWei('22'), ticket.address).returns()
+        await prizePool.withdrawWithTimelockFrom(wallet2._address, toWei('22'), ticket.address)
+  
+        // Only first deposit is unlocked
+        await prizePool.setCurrentTime('15')
+  
+        debug('GOT HERE 3')
+  
+        // expect the redeem && transfer for only the unlocked amount
+        await cToken.mock.redeemUnderlying.withArgs(toWei('11')).returns('0')
+        await token.mock.transfer.withArgs(wallet._address, toWei('11')).returns(true)
+        await comptroller.mock.afterSweepTimelockedWithdrawal.withArgs(wallet._address, wallet._address, toWei('11')).returns()
+  
+        debug('GOT HERE 4')
+  
+        // Let's sweep
+        let result = await prizePool.sweepTimelockBalances([wallet._address, wallet2._address])
```

Comment:
I was trying to assert the existence and non-existence of multiple events in the same transaction, so I extracted it to a variable.  However, the *negation* of an event didn't work (i.e. `.not.to.emit...`) so i scrapped it.

This can go back to the more succinct form